src/stan/services/sample/defaults.hpp

```diff
@@ -0,0 +1,516 @@
+#ifndef STAN_SERVICES_SAMPLE_DEFAULTS_HPP
+#define STAN_SERVICES_SAMPLE_DEFAULTS_HPP
+
+#include <stdexcept>
+#include <string>
+
+namespace stan {
+  namespace services {
+    namespace sample {
+
+      /**
+       * Number of sampling iterations.
+       */
+      struct num_samples {
+        /**
+         * Description of num_samples.
+         *
+         * @returns description
+         */
+        static std::string description() {
+          return "Number of sampling iterations.";
+        }
+
+        /**
+         * Validates num_samples; num_samples must be greater than or
+         * equal to 0.
+         *
+         * @param[in] num_samples argument to validate
+         * @throw std::invalid argument if num_samples < 0
+         */
+        static void validate(int num_samples) {
+          if (!(num_samples >= 0))
+            throw std::invalid_argument("num_samples must be greater "
+                                        "than or equal to 0.");
+        }
+
+        /**
+         * Default num_samples: 1000.
+         *
+         * @returns default
+         */
+        static int default_value() {
+          return 1000;
+        }
+      };
+
+      /**
+       * Number of warmup iterations.
+       */
+      struct num_warmup {
+        /**
+         * Description of num_warmup.
+         *
+         * @returns description
+         */
+        static std::string description() {
+          return "Number of warmup iterations.";
```

Comment:
"Number of samples (from 'num_samples') to use for warmup."  Otherwise it's not clear that this is a subset of num_samples.